File: app_project/backend/app/schemas/media/video.py
from pydantic import BaseModel
from datetime import datetime

class VideoBase(BaseModel):
    title: str
    description: str | None = None
    youtube_url: str
    is_active: bool = True

class VideoCreate(VideoBase):
    thumbnail: bytes | None = None

class VideoUpdate(VideoBase):
    thumbnail: bytes | None = None

class VideoResponse(VideoBase):
    id: int
    published_at: datetime
    created_at: datetime
    updated_at: datetime
    thumbnail: bytes | None = None

    class Config:
        from_attributes = True

File: app_project/frontend/src/models/media/video.ts
export interface VideoBase {
  title: string;
  description?: string | null;
  youtube_url: string;
  is_active: boolean;
}

export interface VideoCreate extends VideoBase {
  thumbnail?: File | null;
}

export interface VideoUpdate extends Partial<VideoBase> {
  thumbnail?: File | null;
}

export interface VideoResponse extends VideoBase {
  id: number;
  thumbnail?: string | null; 
  published_at: string;
  created_at: string;
  updated_at: string;
}
